File: Reverse a string using Stack - GFG/reverse-a-string-using-stack.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;
char* reverse(char *str, int len);
int main()
{
    long long int t;
    cin>>t;
    while(t--)
    {
        char str[10000];
        cin>>str;
        long long int len=strlen(str);
        char *ch=reverse(str,len);
        for(int i=0;i<len;i++)
        {
            cout<<ch[i];
        }
        cout<<endl;
    }
        return 0;
}


// } Driver Code Ends


//return the address of the string
char* reverse(char *s, int n)
{
    //code here
    stack<char>res;
    for(int i = 0 ; i < n ; i++){
        res.push(s[i]);
    }
    int j = 0 ;
    while(!res.empty()){
        s[j++] = res.top();
        res.pop();
    }
    return s;
}

File: Reverse a string using Stack - GFG/reverse-a-string-using-stack.py
#{ 
 # Driver Code Starts

# } Driver Code Ends
def reverse(s):
    
    #Add code here
    stack = []
    res = ""
    for i in s:
        stack.append(i)
    for i in range(len(stack)):
        res +=stack.pop()
    return res
    
    
#{ 
 # Driver Code Starts.
if __name__=='__main__':
    t=int(input())
    for i in range(t):
        str1=input()
        print(reverse(str1))
# } Driver Code Ends